packages/ua-devtools-evm-hardhat/src/oapp/typescript/typescript.ts

```diff
@@ -0,0 +1,495 @@
+        )
+    } else if (receiveDefaultUlnConfig == null) {
+        throw new Error(
+            `An error occurred while generating LayerZero Config: No Default Receive ULN Config for pathway ${fromNetwork} → ${toNetwork}`
+        )
+    }
+
+    return factory.createPropertyAssignment(
+        factory.createIdentifier(CONFIG),
+        factory.createObjectLiteralExpression([
+            createSendLibraryConfig(sendDefaultLibrary),
+            createReceiveLibraryConfig(receiveDefaultLibrary),
+            createSendConfig(sendDefaultExecutorConfig, sendDefaultUlnConfig),
+            creatReceiveConfig(receiveDefaultUlnConfig),
+        ])
+    )
+}
+
+/**
+ * Generates a default LayerZero configuration for the selected networks and contract.
+ *
+ * @param {string[]} selectedNetworks An array of network names.
+ * @param {string} contractName The name of the contract.
+ * @returns {Promise<NodeArray<Statement>>} A promise that resolves to a NodeArray containing generated LayerZero configuration statements.
+ */
+export const generateLzConfig = async (
+    selectedNetworks: string[],
+    contractName: string
+): Promise<NodeArray<Statement>> =>
+    factory.createNodeArray([
+        createEndpointImportDeclaration(),
+        ...createContractVariables(selectedNetworks, contractName),
+        await createContractsAndConnections(
+            new Map(
+                selectedNetworks.map((network) => [
+                    network,
+                    factory.createIdentifier(normalizeIdentifierName(network) + CONTRACT),
+                ])
+            )
+        ),
+    ])
+
+/**
+ * Retry policy function that retries a given asynchronous operation with exponential backoff strategy.
+ * @template T - The type of the result returned by the asynchronous operation.
+ * @param {() => Promise<T>} fn - The asynchronous operation to be retried.
+ * @param {number} [maxAttempts=3] - The maximum number of attempts.
+ * @param {number} [baseDelay=1000] - The base delay in milliseconds before the first retry.
+ * @param {number} [maxDelay=15000] - The maximum delay in milliseconds between retries.
+ * @returns {Promise<T>} - A promise resolving to the result of the operation if successful.
+ * @throws {Error} - Throws an error if all attempts fail.
+ */
+async function basicRetryPolicy<T>(
+    fn: () => Promise<T>,
+    maxAttempts: number = 3,
+    baseDelay: number = 1000,
+    maxDelay: number = 15000
+): Promise<T> {
+    const operation = async () => {
+        return await fn()
+    }
+
+    const backoffOptions = {
+        numOfAttempts: maxAttempts,
+        startingDelay: baseDelay,
+        delayFirstAttempt: true,
+        maxDelay: maxDelay,
+        factor: 2,
+        randomisationFactor: 0.5,
+    }
+
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
+        try {
+            logger.verbose(`Attempt ${attempt + 1}/${maxAttempts}`)
+            return await backOff(operation, backoffOptions)
```

Comment:
Per offline discussion, you can just embed the logging in a an optional retry function.